### PR TITLE
all kind of fixes + better build

### PR DESCRIPTION
**Pull request by SimonDanisch**
* use bindeps for build
* better teardown and initialization
* various fixes

### CODE REVIEW

File: src/CLFFT.jl [hunk 1]
```diff
@@ -61,23 +61,25 @@ typealias clfftTypeSingle Union{Type{Float32},Type{Complex64}}
 
 typealias PlanHandle Csize_t
 
-
-type Plan{T<:clfftNumber}
+function free(x) end
```
Comment (vchuravy) [inline src/CLFFT.jl:64]: This definition is redundant :)

File: src/CLFFT.jl [hunk 2]
```diff
@@ -61,23 +61,25 @@ typealias clfftTypeSingle Union{Type{Float32},Type{Complex64}}
...
-            x.id[1] = C_NULL
-        end)
+        finalizer(p, free)
         return p
     end
 end
+const global is_initialized = Ref(false)
 
+function free(x::Plan)
+    if x.id[1] != 0 && is_initialized[]
```
Comment (vchuravy) [inline src/CLFFT.jl:78]: So it was a free after deinitialization?

File: src/api.jl [hunk 3]
```diff
@@ -2,50 +2,30 @@ module api
...
 import OpenCL.cl
 
-@static if is_unix() const libclfft = "libclFFT" end
-@static if is_windows() const libclfft = "clFFT" end
-
+depsfile = joinpath(dirname(@__FILE__), "..", "deps", "deps.jl")
+if isfile(depsfile)
+    include(depsfile)
+else
+    error("CLFFT not properly installed. Please run Pkg.build(\"CLFFT\") then restart Julia.") # now that this is decoupled from images, should this be an error?
```
Comment (vchuravy) [inline src/api.jl]: the comment seems odd

File: test/runtests.jl [hunk 4]
```diff
@@ -169,7 +169,7 @@ facts("Example FFT Double") do
 end
 
 facts("2D FFT Inplace") do
```
Comment (vchuravy) [inline test/runtests.jl:171]: maybe transition to testsets?.

**Review by vchuravy — APPROVED** [review]
Looks good :) Maybe some more squashing, bu otherwise LGTM.

File: src/api.jl (hunk 3 above)
Comment (SimonDanisch) [reply src/api.jl]: hahaha, copy and paste...

File: test/runtests.jl (hunk 4 above)
Comment (SimonDanisch) [reply test/runtests.jl:171]: in another pr ;)

**Pull request merged**